torch_geometric/nn/pool/mesh_pool.py

```diff
@@ -0,0 +1,266 @@
+import torch
+import torch.nn as nn
+from threading import Thread
+from ..mesh.mesh_union import MeshUnion
+from ..mesh.mesh_edge_padding import get_meshes_edge_index
+import numpy as np
+from heapq import heappop, heapify
+
+
+class MeshPool(nn.Module):
+    r"""The Mesh Pooling operator from the `"MeshCNN: A Network with an Edge"
+     <https://arxiv.org/abs/1809.05910>`_ paper
+    This class implements the pooling operation on a Mesh structure.
+    We define mesh pooling as a series of edge collapse operations, where each
+    edge collapse converts five edges into two.
+    Edge collapse: For an edge :math:`\mathbf{e}` with edge neighbors
+    :math:`\left(a, b \right)` and :math:`\left(c,d \right)`, which define two
+    triangles, respectively, we create two new edge features according to:
+    .. math::
+    `\mathbf{MeshConv(p)_i} = \mathbf{avg}\left ( \mathbf(a)_i, \mathbf(b)_i,
+                                                  \mathbf(e)_i \right )`
+    `\mathbf{MeshConv(q)_i} = \mathbf{avg}\left ( \mathbf(c)_i, \mathbf(d)_i,
+                                                  \mathbf(e)_i \right )`
+    where :math:`\mathbf{p}` and :math:`\mathbf{q}` are the resulting two new
+    feature vectors, and :math:`\mathbf{i}` represents the feature channel
+    index.
+
+    The edge collapse order is according to L2-norm of the edge features, where
+    the smallest norms collapse first. Edge collapse stops according to the
+    target pooling resolution.
+
+    Args:
+        target_resolution (int): number of edges after pooling operation
+        multi_thread (bool, optional): use multi-thread for the pooling process
+                                       (default - False)
+    """
+
+    def __init__(self, target_resolution: int, multi_thread: bool = False):
+        super(MeshPool, self).__init__()
+        self.__out_target = target_resolution
+        self.__multi_thread = multi_thread
+        self.__fe = None
+        self.__updated_fe = None
+        self.__meshes = None
+        self.__merge_edges = [-1, -1]
+
+    def __call__(self, fe, meshes):
+        return self.forward(fe, meshes)
```

Comment:
Maybe I am missing something, but the `__call__` function in `nn.Module` calls `forward`, do you have a specific reason for over ridding it.
```suggestion
```